### PR TITLE
Labyrinth 🕹 - Sofia Vaz Sousa & Jessica Panditha

### PR DESCRIPTION
**Pull request by sofiavazs**
**_Link to my deployed project:_**
https://labyrinth-sj.netlify.app/

**_My reflections on how this week's project turned out:_**

**Sofia:** 
I'm really really proud how the project turned out, on Monday what it seemed an impossible painful task has turned into the most challenging and fun project that helped me so much to battle and learn readux and it's flow! I'm happy that we were able to achieve some of the stretch goals too! 

**Jessica:**
This project was quite difficult and it was hard to understand what we were supposed to do and how to use POST request and thunks to create the labyrinth and we had some blockers along the way. But in the end I think we managed to create something that we are very proud of

Things I'd like to have clarified or explained in more detail:
None.

### CODE REVIEW

File: code/src/StyledComponents/GlobalStyles.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  border: 1px inset #fd20e3;
+  background: #fd20e3;
+}
+  @media (min-width: 768px) {
+    margin:40px;
+  }
+`;
+
+export const Hint = styled.p`
+  font-size:9px;
```
Comment (MindstormingAB) [inline code/src/StyledComponents/GlobalStyles.js:65]: Avoid too small font-size (not under 12px) 👓

**Review by MindstormingAB — CHANGES_REQUESTED** [review]
Sofia and Jessica, well done with this week's project!
The game works very well and is easy to use. The design is really nice, simple and efficient. Cool with different background colors! We liked it!
You showed in the project that you understand redux principles and can use thunks.
You reached the blue level requirements and parts of the red and black level requirements.
However, we'd like you to fix the following before the project can be approved:
- revise indentation, semi-colons, spaces... according to Technigo's code guidelines
- increase the font-size of your hints
- make sure that you display each direction's hint in GameContinue.js

We were grateful about your comments because we sometime had a hard time following the rendering logic. Everything works well but you might have done things more complicated than they needed to be. 💐❤ 
Keep up the good work! 💪

File: code/src/StyledComponents/CreatePlayerStyling.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+};
+`;
+
+export const InputText = styled.input`
+  text-align:center;
+  width:80%;
+  line-height:50px;
+  border-radius:16px;
+  border:none;
+`;
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply code/src/StyledComponents/CreatePlayerStyling.js]: Thanks for the tip, really difficult to write css in styled components!

File: code/src/components/GameContinue.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  };
+
+  return(
+      <Container>
+        <BoxShadow>
+          <Text>{currentState.labyrinth.game.description}</Text>
+            {actions.map((action)=> 
+          <Button key= {action.description} onClick={() => onGameContinue(action.direction)}> Go {action.direction}</Button>
+          )}
+          <Hint> *HINT* {currentState.labyrinth.game.actions[0].description}</Hint>
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply code/src/components/GameContinue.js]: We opted not to show both :p

File: code/src/components/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const HomePage = () => {
+  const isLoading = useSelector((state) => state.ui.isLoading);
+
+  return (
+    <>
+      {isLoading ? <LoadingPage /> : <CreatePlayer/>}
+    </>
+  );
+};
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply code/src/components/HomePage.js:20]: Thanks, understand what you mean but it made sense for us to have a flow and rendering components as we went along. But maybe one of the code coaches can comment what's the best approach in this case! 😊

File: code/src/StyledComponents/GlobalStyles.js (hunk 1 above)
Comment (sofiavazs) [reply code/src/StyledComponents/GlobalStyles.js]: It was supposed to be a hint! 😈 fixed :p

File: code/src/reducers/labyrinth.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import  { createSlice } from '@reduxjs/toolkit';
+
+export const labyrinth = createSlice({
+    name: 'labyrinthGame',
+    initialState: {
+        username:"",
+        game: { 
+          actions: [],
```
[the remark this answers is not in the document]
Comment (sofiavazs) [reply code/src/reducers/labyrinth.js:8]: Not at all! It was actually quite helpful to create the end page ;)

**Comment by sofiavazs** [comment]
Thank you for the feedback! We've now made the changes requested 😊

File: code/src/components/CreatePlayer.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  return(
+    <Container>
+      <MainHeader>Welcome User!</MainHeader>
+      <BoxShadow>
+        <Form onSubmit={handleSubmitPlayerName}>
+         <InputText
+            type = "text"
+            value = {username}
+            onChange={(event) => handleCreatePlayer(event)}
```
Comment (puj) [inline code/src/components/CreatePlayer.js:59]: The only thing that is not clear here is the fact that `handleCreatePlayer` actually updates the username.  
Also, due to the naming, one could misunderstand that a player is created for each keypress in the textbox.

File: code/src/components/GameContinue.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      return background.backgroundColor = "#344f77";
+    } else if (currentCoordinates === "0,3") {
+      return background.backgroundColor = "#2e4f80";
+    } else if (currentCoordinates === "1,3") {
+        return background.backgroundColor = "#910591";
+    } else {
+      return background.backgroundColor = "#000000";
+    }
+  };
+  window.onload = backgroundChange();
```
Comment (puj) [inline code/src/components/GameContinue.js:45]: Really nice user of color progression.  Is there a way to make a solution for 100 rooms without making 100 if/else statements?

File: code/src/reducers/gameData.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import { labyrinth } from './labyrinth';
+import { ui } from './ui'; 
+
+const START_URL = "https://wk16-backend.herokuapp.com/start";
```
Comment (puj) [inline code/src/reducers/gameData.js:4]: Yes please :)

File: code/src/reducers/gameData.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	  })
+    .then(res => res.json())
+    .then(json => {
+      dispatch(labyrinth.actions.continueGame(json));
+      dispatch(ui.actions.setLoading(false));
+    });
+  };
+};
+
+const CONTINUE_URL = "https://wk16-backend.herokuapp.com/action";
```
Comment (puj) [inline code/src/reducers/gameData.js:23]: You can group this with `START_URL`

**Review by puj — APPROVED** [review]
Really nice job! You exhibited understanding of quite a few deep concepts.  I like the thought process behind the naming and most of your logic.  Well done.  Clean project!

File: code/src/components/GameContinue.js (hunk 7 above)
Comment (sofiavazs) [reply code/src/components/GameContinue.js:45]: You got me thinking about this one and if I could create an array with colours and create a function to randomise them on window load 😬 (they wouldn't be specific to the coordinates but could be a nice solution 🙈)

**Pull request closed without merging**